File: starters_example/src/test/java/com/example/service/DBExampleServiceTest.java
package com.example.service;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.example.base.TestBase;
import com.example.model.UserMain;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;

class DBExampleServiceTest extends TestBase {

  private final DBExampleService dbExampleService;

  @Autowired
  DBExampleServiceTest(DBExampleService dbExampleService){
    this.dbExampleService = dbExampleService;
  }

  @Test
  void findUserMainByNameTest(){
    UserMain userMain = dbExampleService.getUserMainByName("Gary");
    assertEquals("Gary", userMain.getName());
  }
}

File: example-kotlin/src/test/kotlin/com/example/service/DBExampleServiceTest.kt
package com.example.service

import com.example.base.TestBase
import io.kotest.matchers.shouldBe

/**
 * [DBExampleService] 整合測試：驗證可依名稱查得使用者主要資料。
 */
class DBExampleServiceTest(
    private val dbExampleService: DBExampleService,
) : TestBase({

    test("findUserMainByNameTest") {
        val userMain = dbExampleService.getUserMainByName("Gary")
        userMain?.name shouldBe "Gary"
    }
})
